cli/src/pcluster/validators/cluster_validators.py

```diff
@@ -182,6 +183,63 @@ def validate_accelerator_requirements(
         self.validate_accelerator_count(compute_resource_name, instance_types_info)
         self.validate_accelerator_manufactures(compute_resource_name, instance_types_info)
 
+    def validate_efa_requirements(
+        self,
+        compute_resource_name: str,
+        instance_types_info: Dict[str, InstanceTypeInfo],
+        efa_enabled: bool,
+    ):
+        """Check if the instance types have a uniform support for EFA.
+
+        Validation Failure is expected if EFA is ENABLED and at least one instance type defined in the compute resource
+        DOES NOT support EFA.
+        """
+        if efa_enabled:
+            all_instance_types = set(instance_types_info.keys())
+            instance_types_without_efa_support = {
+                instance_type_name
+                for instance_type_name, instance_type_info in instance_types_info.items()
+                if not instance_type_info.is_efa_supported()
+            }
+
+            # If all the instance types have EFA support, `instance_types_without_efa_support` should be empty
+            # --> No failure expected
+            # If all the instance types DO NOT have EFA support, `instance_types_without_efa_support` should be the same
+            # as `all_instance_types` (Set difference of 0) --> No failure expected
+            # If there is a mixed support for EFA, `instance_types_without_efa_support` & `all_instance_types`
+            # will have different instance types (Set difference greater than 0) --> Validation Failure expected and
+            # a message with the instance types that DO NOT support EFA is included
+            if instance_types_without_efa_support and len(all_instance_types - instance_types_without_efa_support) > 0:
+                self._add_failure(
+                    (
+                        "Instance types ({0}) in Compute Resource {1} do not support EFA.".format(
```

Comment:
`types` -> `type(s)`

Reply:
what about adding the reason of the failure here? Something like "and cannot be launched because you have enabled EFA [in CR configuration]"